src/ansible_navigator/app.py

```diff
@@ -78,46 +85,77 @@ def app(self) -> AppPublic:
         raise AttributeError("app passed without args initialized")
 
     @staticmethod
-    def _copy_args(args):
-        """deepcopy the args, but un mount the cache first
+    def _copy_args(args: ApplicationConfiguration) -> ApplicationConfiguration:
+        """Deepcopy the args.
+
+        Note: Unmount the collection doc cache (cdc) first
         the cdc will get mounted if the child needs it
         in parse and update
+
+        :param args: the current application configuration
+        :return: A copy of the current application configuration
         """
         args.internals.collection_doc_cache = C.NOT_SET
         return deepcopy(args)
 
     def _prepare_to_run(self, app: AppPublic, interaction: Interaction) -> None:
+        """Prepare for action run.
+
+        An action can call this to prior to running. This
+        will set the scroll to zero and store the state
+        of the UI so it can be restored later.
+
+        :param app: The instance of the action
+        :param interaction: The current interaction from the ui
+        """
         self._calling_app = app
         self._interaction = interaction
         self._interaction.ui.scroll(0)
         self._previous_scroll = interaction.ui.scroll()
         self._previous_filter = interaction.ui.menu_filter()
 
-    def _prepare_to_exit(self, interaction) -> None:
-        """Prior to exiting an app can call this to clean up"""
+    def _prepare_to_exit(self, interaction: Interaction) -> None:
+        """Prepare for action exit.
+
+        An action can call this after running to clean up and
+        restore the state of the UI prior to the action being
+        invoked.
+
+        :param interaction: The current interaction from the ui
+        """
         interaction.ui.scroll(self._previous_scroll)
         interaction.ui.menu_filter(self._previous_filter)
         interaction.ui.scroll(0)
 
     def parser_error(self, message: str) -> Tuple[None, None]:
-        """callback for parser error
+        """Store the argparse parser error.
+
+        This is used as a callback for argparse.
 
         :param message: A message from the parser
-        :type message: str
+        :return: Nothing
```

Comment:
Is this method actually used?

```
main [ansible/winston] $ rg parser_error
src/ansible_navigator/app.py
41:        self._parser_error: str = ""
102:    def parser_error(self, message: str) -> Tuple[None, None]:
108:        self._parser_error = message
```

In any case, it's not returning `Nothing`, it's returning a two element tuple with both components being the value `None`.